### PR TITLE
simplify code

### PR DESCRIPTION
**Pull request by xavier-figueroa**
Inline variables
Simplify string method calls
Collapse catch blocks
Use diamond operator
Use Math.min/max whenever possible
Simplify array method calls
Remove exceptions that are never thrown
Prefer Files.newOutputStream over new FileOutputStream, has better performance
StringBuilder over StringBuffer

### CODE REVIEW

File: src/com/lightdev/app/shtm/SHTMLEditorKitActions.java [hunk 1]
```diff
@@ -1974,7 +1972,7 @@ static class PrintAction extends AbstractAction implements SHTMLAction {
         static {
             Method printMethod = null;
             try {
-                printMethod = JTextComponent.class.getMethod("print", new Class[] {});
+                printMethod = JTextComponent.class.getMethod("print");
```
Comment (dpolivaev) [inline src/com/lightdev/app/shtm/SHTMLEditorKitActions.java]: Keep parameters because there is more than one method with name "print"

Comment (xavier-figueroa) [reply src/com/lightdev/app/shtm/SHTMLEditorKitActions.java]: done

**Comment by dpolivaev** [comment]
Please don't push force commits because then I have to review EVERYTHING again and it costs me a lot of my time.
You can always add new commits but please keep already reviewed commits on your branch

File: src/com/lightdev/app/shtm/CombinedAttribute.java [hunk 2]
```diff
@@ -238,9 +239,7 @@ public int getOppositeSide(final int side) {
     private void copyValues(final String[] v) {
         switch (v.length) {
             case 1:
-                for (int i = 0; i < values.length; i++) {
-                    values[i] = v[0];
-                }
+                Arrays.fill(values, v[0]);
```
Comment (dpolivaev) [inline src/com/lightdev/app/shtm/CombinedAttribute.java]: Using Arrays.fill for length 1 looks strange

Comment (xavier-figueroa) [reply src/com/lightdev/app/shtm/CombinedAttribute.java]: you're right, it was just to hide the for loop. I guess we can replace the for loop with  `values[0] = v[0];`

**Pull request merged**